src/main/database.cpp

```diff
@@ -284,6 +292,7 @@ void DatabaseInstance::Configure(DBConfig &new_config) {
 	if (!config.default_allocator) {
 		config.default_allocator = Allocator::DefaultAllocatorReference();
 	}
+	config.custom_buffer_manager = std::move(new_config.custom_buffer_manager);
```

Comment:
This probably needs a `Duplicate` or `Copy` method, we don't want to modify the config that's passed in